Formula/bootloadhid.rb

```diff
@@ -0,0 +1,19 @@
+class Bootloadhid < Formula
+  desc "HID-based USB bootloader for AVR microcontrollers"
+  homepage "https://www.obdev.at/products/vusb/bootloadhid.html"
+  url "https://www.obdev.at/downloads/vusb/bootloadHID.2012-12-08.tar.gz"
+  version "2012-12-08"
+  sha256 "154e7e38629a3a2eec2df666edfa1ee2f2e9a57018f17d9f0f8f064cc20d8754"
+
+  depends_on "libusb-compat"
+
+  def install
+    Dir.chdir "commandline"
+    system "make"
+    bin.install "bootloadHID"
+  end
+
+  test do
+    assert File.exist?("#{bin}/bootloadHID")
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
Thanks. I've changed the test to hopefully something a little better.